src/services/browser/BrowserSession.ts

```diff
@@ -1,88 +1,131 @@
 import * as vscode from "vscode"
 import * as fs from "fs/promises"
 import * as path from "path"
-import { Browser, Page, ScreenshotOptions, TimeoutError, launch } from "puppeteer-core"
-// @ts-ignore
-import PCR from "puppeteer-chromium-resolver"
+import { Browser, Page, ScreenshotOptions, TimeoutError, launch, connect } from "puppeteer"
 import pWaitFor from "p-wait-for"
 import delay from "delay"
 import { fileExistsAtPath } from "../../utils/fs"
 import { BrowserActionResult } from "../../shared/ExtensionMessage"
-
-interface PCRStats {
-	puppeteer: { launch: typeof launch }
-	executablePath: string
-}
+import axios from 'axios'
 
 export class BrowserSession {
 	private context: vscode.ExtensionContext
 	private browser?: Browser
 	private page?: Page
 	private currentMousePosition?: string
+	private isInteractive: boolean = false
+	private browserPort: string = '7333'
 
 	constructor(context: vscode.ExtensionContext) {
 		this.context = context
 	}
 
-	private async ensureChromiumExists(): Promise<PCRStats> {
-		const globalStoragePath = this.context?.globalStorageUri?.fsPath
-		if (!globalStoragePath) {
-			throw new Error("Global storage uri is invalid")
-		}
+	async launchBrowser(interactive: boolean = false, port?: string) {
+		console.log("launch browser called")
+		this.isInteractive = interactive
 
-		const puppeteerDir = path.join(globalStoragePath, "puppeteer")
-		const dirExists = await fileExistsAtPath(puppeteerDir)
-		if (!dirExists) {
-			await fs.mkdir(puppeteerDir, { recursive: true })
+		// Set browserPort if provided, otherwise use default
+		if (port) {
+			this.browserPort = port
```

Comment:
Does this effectively remove chromium functionality from Roo Cline, replacing it with a more "real" browser interaction functionality?

Reply:
not necessarily, in `package.json` we changed `puppeteer-chromium-resolver` and `puppeteer-core` for the entire `puppeteer` library and it already has a chromium browser as part of it.

That puppeteer chromium is used by default, but if you specify in your prompt `interactive mode` that's when it'll need a "real" browser